Lib/mimetypes.py

```diff
@@ -88,7 +89,15 @@ def add_type(self, type, ext, strict=True):
         If strict is true, information will be added to
         list of standard types, else to the list of non-standard
         types.
+
+        Non-empty extensions that don't start with a '.' are not
+        valid, so a ValueError will be raised if they are
+        specified.
         """
+        if not ext:
+            warnings.warn('Empty extension specified')
```

Comment:
As told on b.p.o, I found this morning a place prooving the empty ext is usefull, so I think now we should just let people use the empty type as a default fallback.
So maybe a ```if ext and not ext.startswith("."):``` then, and tell in the ValueError that it either should start with dot or be empty?